### DISCUSSION TITLE
Can not parse newly created Pull Request

### DISCUSSION THREAD

**Discussion by kimpenhaus** [renovatebot/renovate#11814] [Request Help] [answered]
Hi all,

I am running the renovate-bot selfhosted on the latest docker image against a selfhosted gitea instance (also latest).
I configured to run against a python project with enabledManagers `pip_requirements`.

renovate config.json

```json
    "platform": "gitea",
    "endpoint": "https://git.mydomain.net/api/v1",
    "username": "renovate-bot",
    "password": "<some-password>",
    "token": "<some-gitea-token>",
    "autodiscover": false,
    "repositories": [
        "my-repository"
    ]
```

repository renovate.json

```json
{
    "labels": [
        "dependencies"
    ],
    "automerge": true,
    "unicodeEmoji": false,
    "enabledManagers": [
        "pip_requirements"
    ],
    "gitAuthor": "renovate-bot <some@email.net>"
}
```

After hitting the mysql/mariadb collation problem with emojis and changing the collation to `utf8mb4_general_ci` I am able to create PRs. Which is pretty cool!

My problem is, that I need to close every PR manually and was hoping to have automerge working. But it is not... and I think this might be related to some log entries I am facing:

```json
{
    "name": "renovate",
    "hostname": "7ad653bd60fb",
    "pid": 8,
    "level": 20,
    "logContext": "L1Ztkdq7D",
    "repository": "my-repository",
    "branch": "renovate/py-linq-1.x",
    "err": {
        "message": "Can not parse newly created Pull Request",
        "stack": "Error: Can not parse newly created Pull Request\n    at Proxy.createPr (/usr/src/app/node_modules/renovate/lib/platform/gitea/index.ts:489:15)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at ensurePr (/usr/src/app/node_modules/renovate/lib/workers/pr/index.ts:408:14)\n    at processBranch (/usr/src/app/node_modules/renovate/lib/workers/branch/index.ts:596:33)\n    at writeUpdates (/usr/src/app/node_modules/renovate/lib/workers/repository/process/write.ts:38:17)\n    at update (/usr/src/app/node_modules/renovate/lib/workers/repository/process/extract-update.ts:129:11)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:50:17)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:111:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
    },
    "msg": "Pull request creation error",
    "time": "2021-09-19T23:01:51.927Z",
    "v": 0
}
```

I can't find any further information nor have any clue how to deal with it. The PR itself s created fine (as already said I can merge it manually)

Any ideas what to do... where to look ? Any help is appreciated!

Thanks!
Marcus

**Comment by viceice** [comment, marked as answer]
Do not use password and token auth, so remove the password

**Comment by kimpenhaus** [comment]
@viceice removed it  - but still the same. PR is created successfully but not automerged:

![image](https://user-images.githubusercontent.com/575109/133975787-d06ad4dd-221c-41ae-ab5f-8fe75d3ef055.png)

might this be an issue with the `gitAuthor`?

`"gitAuthor": "renovate-bot <some@email.net>"` I gave the renovate-bot one of my emails... and gitea says that my user (not the bot - which actually is also added as user to gitea) wants to merge...

referring to: https://github.com/renovatebot/renovate/blob/4303063b2ab77e5680d96fa659dd146195f781d7/lib/platform/gitea/index.ts#L89

**Comment by kimpenhaus** [comment]
hm - that's really strange ... 

that's the token I am using:

![image](https://user-images.githubusercontent.com/575109/133980262-1e575c32-2cc4-4dfe-af6b-d1e58a320755.png)

that's my config.json of the renovate-bot docker:

![image](https://user-images.githubusercontent.com/575109/133980443-92faeb5e-4989-4670-ac77-2bbcbd68a9a2.png)

when querying the gitea api for the create PR:

![image](https://user-images.githubusercontent.com/575109/133980563-42e23792-4df2-4e5b-87b2-c2906773878c.png)

so it looks like the user created the PR is `kimpenhaus` and not `renovate-bot` 🤔 

`renovate-bot` ist just collaborator - might that be an issue? The repo is inside an organization `private` and `kimpenhaus` is the owner. But I guess I have to ask over at gitea as this is theirs internals - right?

![image](https://user-images.githubusercontent.com/575109/133980789-f98c3380-55e3-4082-b374-0a40472605bd.png)

**Reply by kimpenhaus** [reply]
https://discourse.gitea.io/t/pullrequest-user-creator/3796 - I aksed at the gitea community

**Reply by viceice** [reply]
maybe you used a wrong token on first run? can you try to close and rename the pr and see if renovate created a new pr with bot account as author?

I'm using renovate with gitea and a dedicated bot account too and never had those issues. Maybe you have some additional environment variables with start with `RENOVATE_` ? Those will be parsed as config options. So please do a debug run and paste the sanitized logs here.

**Reply by kimpenhaus** [reply]
Hell yeah - I got it working...

So I started over - created new repo added some files and tried again... only difference to my existing repo was the `Signing Verification Settings` of the repo (by default it is: `Default Trust Model` - I changed it to `Collaborator+Committer: Trust signatures by collaborators which match the committer`)

Ran the docker ... and ... status is yellow ... okay fixed it by ignoring tests and voilà ... PR created and auto merged.

So I've changed the `Signing Verification Settings` of my existing repo to the same settings and it is working 🤷🏼‍♂️ 

Can you verify that your settings are similar? Actually I don't have any clue what effect that should have...

**Reply by viceice** [reply]
nope, i've different models, depends on repo and all works as expected. 🤷‍♂️

**Reply by kimpenhaus** [reply]
strange ...

nevertheless seems to be working now. 
thanks a lot for your time - really appreciate! 🙏🏼 

also not sure where to put the `answer` mark... have still no clue why it's working now 😁

**Reply by viceice** [reply]
i assume you've played with the wrong PAT the first time, so the PR's are opened with wrong user. 😉
